catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java

```diff
@@ -13,139 +13,756 @@
+                        for (Callback callback : callbacks.values()) {
+                            try {
+                                if (standingQueryData.size() > 0) {
+                                    LOGGER.trace("Notifying callback that results are available: {}",
+                                            standingQueryData.size());
+                                    callback._notify(org.codice.alliance.nsili.common.UCO.State.RESULTS_AVAILABLE,
+                                            get_request_description());
+                                }
+                            } catch (InvalidInputParameter invalidInputParameter) {
+                                LOGGER.warn("Unable to notify callback", invalidInputParameter);
+                            } catch (ProcessingFault processingFault) {
+                                LOGGER.warn("Unable to notify callback", processingFault);
+                            } catch (SystemFault systemFault) {
+                                LOGGER.warn("Unable to notify callback", systemFault);
+                            } catch (Exception e) {
+                                failedCallbacks.add(callback);
+                            }
+                        }
+
+                        failedCallbacks.stream()
+                                .forEach(c -> freeCallback(c));
+                    }
+                    lastCompletedExecutionTime = System.currentTimeMillis();
+                }
+
+                //Don't sleep if more results available. Need client to pick up results as fast
+                //as possible to catch up.
+                if (!executionThread.hasMoreResultsAvailOnLastQuery()) {
+                    try {
+                        Thread.sleep(updateRate);
+                    } catch (Exception ignore) {
+                    }
+                }
+            }
+
+            synchronized (callbackLockObj) {
+                callbacks.clear();
+            }
+        }
+
+        protected DAGQueryResult getData(long queryTime) {
+            DAGQueryResult result = null;
+
+            List<Result> catalogResults = new ArrayList<>();
+
+            Filter parsedFilter = bqsFilter;
+            if (!moreResultsAvailOnLastQuery && queryTime > 0) {
+                parsedFilter = filterBuilder.allOf(bqsFilter,
+                        filterBuilder.attribute(Metacard.MODIFIED)
+                                .is()
+                                .after()
+                                .date(new Date(queryTime)));
+            }
+
+            catalogQuery = new QueryImpl(parsedFilter);
+            catalogQuery.setRequestsTotalResultsCount(true);
+            catalogQuery.setPageSize(pageSize);
+            if (moreResultsAvailOnLastQuery) {
+                catalogQuery.setStartIndex(startIndex);
+            }
+
+            QueryRequestImpl catalogQueryRequest = new QueryRequestImpl(catalogQuery);
```

Comment:
Looks like right now that is defaulted to Local query. Good catch. I didn't map the enterpriseSearch option through to this.